### PR TITLE
Feature/vulkan pipeline

### PR DESCRIPTION
**Pull request by codeplay9800**
+ Created and Integrated Swap Chain Class
+ Refactored Codebase 
(There is still an empty function named acquire in the swap chain class. Thought we might need it in the future)

### CODE REVIEW

File: src/RenderContext.cpp [hunk 1]
```diff
@@ -52,37 +52,15 @@ RenderContext::RenderContext() {
...
 }
 
 RenderContext::~RenderContext() {
-  vkb::destroy_swapchain(swapchain_);
   spdlog::debug("Destroyed swapchain");
   vkb::destroy_device(device_);
   spdlog::debug("Destroyed logical device");
-  vkDestroySurfaceKHR(instance_.instance, surface_,
-                      instance_.allocation_callbacks);
   spdlog::debug("Destroyed Vulkan surface");
```
Comment (shineyruan) [inline src/RenderContext.cpp]: Please also remove line 61. Logging is redundant for destroying Vulkan surfaces.

File: src/Renderer.cpp [hunk 2]
```diff
@@ -9,25 +9,18 @@ namespace volume_restir {
...
 Renderer::Renderer() {
   render_context_ = std::make_unique<RenderContext>();
+  CreateSwapChain();
   InitQueues();
   CreateRenderPass();
   CreateGraphicsPipeline();
   CreateFrameResources();
   CreateCommandPools();
   RecordCommandBuffers();
-  CreateSyncObjects();
```
Comment (shineyruan) [inline src/Renderer.cpp:18]: Please delete this commented function

File: src/Renderer.hpp [hunk 3]
```diff
@@ -51,11 +54,6 @@ class Renderer {
   VkCommandPool graphics_command_pool_;
```
Comment (shineyruan) [inline src/Renderer.hpp:54]: Shall we move `swapchain_image_views_` and `swapchain_images_` inside Swapchain class just as project 5 code does?

**Review by shineyruan — CHANGES_REQUESTED** [review]

File: src/Renderer.hpp (hunk 3 above)
Comment (codeplay9800) [reply src/Renderer.hpp:54]: For some reason the image views still reside in the rendered class in project 5

Comment (shineyruan) [reply src/Renderer.hpp:54]: For some reason you have `SwapChain::vkSwapChainImages` but you neither created it nor used it, and we're still using `Renderer::swapchain_images_` to create frame resources. Could you fix this?

**Review by shineyruan — APPROVED** [review]
LGTM

**Pull request merged**